RHTE-LAB.adoc

```diff
@@ -10,30 +10,482 @@ This repo should act as a resource for cluster administrators to begin the proce
-* Configuring authentication and authorization
-* Configuring namespace creation
-* Setting up quotas & limits for applications
-* Applying automated certificate management for applications
-* Deploying initial workloads
-* Deploying custom dashboards and setting up alerts
 
 Note that the purpose of this lab is not to teach how each of the components works individually, but to show how one can employ a common automation strategy to a myriad of different configurations.
 
 Let's get started...
 
+Provisioning your Lab Environment
+---------------------------------
+
+For Red Hat Tech Exchange, the GUID grabber link to get your lab environment is:
+
+https://www.opentlc.com/gg/gg.cgi?profile=generic_rhte
+
+This lab code is "A2031", the activation code will be provided during the event.
+
+If you are running through the lab on your own then you may provision an "OpenShift 4 Configuration" lab environment from https://labs.opentlc.com/.
+
 :numbered:
 
-Setting up a GitOps Workflow
------------------------------
+GitOps Workflow for a Simple Use Case
+-------------------------------------
 
 _GitOps_ is a form of Infrastructure as Code practice where all of the configurations that define a system are managed in a git repository, and automatically and idempotently applied to that system each time commits are made to the repository.
 For this lab, we are going to enable such a workflow using an Operator called link:https://github.com/KohlsTechnology/eunomia[Eunomia] and link:https://github.com/redhat-cop/openshift-applier[OpenShift Applier], an automation framework for OpenShift.
 Eunomia provides a workflow for watching git repos, triggering actions when new commits are detected.
 The action, in this case, will be a link:https://kubernetes.io/docs/tasks/job/[Kubernetes Job] that executes Applier.
 
+You will begin by configuring a simple and common task for OpenShift clusters, exposing the cluster's integrated image registry.
+First you will configure a registry route manually and then build up to GitOps step by step.
+
+Applying a Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+. Check the initial state of the container image registry.
++
+In your newly installed OpenShift cluster you will find the container image registry running in the `openshift-image-registry` project namespace.
+There is an `image-registry` service, which is used for image access within the cluster, but there is no route for external access.
++
+[subs=+quotes]
+--------------------------------------------------------------------------------
+$ **oc get service -n openshift-image-registry**
+NAME             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+image-registry   ClusterIP   172.30.151.126   <none>        5000/TCP   37m
+$ **oc get route -n openshift-image-registry**
+No resources found.
+--------------------------------------------------------------------------------
+
+. Use the `oc expose` command to create a route for the service:
```

Comment:
It's a common practice to expos the OpenShift registry outside of the cluster so that CI/CD pipelines can link:http://v1.uncontained.io/playbooks/continuous_delivery/image_promotion.html[move images between clusters as a form of Promotion]. We can do this manually using the `oc expose` command.